modules/ROOT/pages/partner-connector-upgrade.adoc

```diff
@@ -0,0 +1,417 @@
+
+[source,java,linenums]
+----
+@ExternalLib(name = "MySQL JDBC Driver",
+description = "A JDBC driver that supports connection to the MySQL Database",
+nameRegexpMatcher = "(.*)\\.jar",
+requiredClassName = "com.mysql.jdbc.Driver",
+coordinates = "mysql:mysql-connector-java:5.1.44")
+public class MySqlConnectionProvider implements ConnectionProvider<Connection> {
+  //
+}
+----
+
+You must update your external libraries to a version that is compatible with Java 17. 
+
+==== Review Your Code
+
+Even though you compile your custom connector with Java 8, ensure you can run your custom connector with Java 17. Reviewing your code is important because there are some restrictions that are flagged as warnings in Java 11 but are flagged as failures and can’t be disabled in Java 17. For example, reflection is more restricted in Java 17, so you must review your code to ensure there are no illegal actions. 
+
+Learn about the different changes in Java 17 to review your code effectively.
+
+===== JDK Migration Guide
+
+See the https://docs.oracle.com/en/java/javase/17/migrate/getting-started.html#GUID-C25E2B1D-6C24-4403-8540-CFEA875B994A[Oracle JDK Migration Guide]. It contains a full list of the changes made in Java 17. Use the guide to assess impact to your code.
+
+===== Mule Runtime Changes
+
+See <<LINK TO MULE RUNTIME>>. It contains a full list of the Java 17 changes made in Mule Runtime. 
+
+* <<Reflection on Java classes>> 
+* <<Serialization>>
+* <<Library upgrades>>
+* <<Changes with running the test suite>>? 
+* <<JPMS>>
+// link to these docs which will be in the Mule Runtime docs
+
+=== Add Missing Code
+
+Because reflection is more restricted in Java 17, API objects now require setters. Previously, API objects and plain old Java objects (POJOs) had default constructors and getters for all properties. Now, API objects and POJOs must also have setters so DataWeave can build outside the connector in the Mule app. 
+
+Constructors and setters are required if your class is instantiated, and getters are required if your class is read. If your class is returned and not instantiated, only getters are required. However, using both getters and setters simplify the validation and certification process. 
+
+[[upgrade-your-rest-sdk-connectors]]
+=== Upgrade Your REST SDK Connectors
+
+Upgrade your REST SDK connectors to Java 17 by using the REST SDK 0.8.0-RC4 version. 
+
+0.8.0-RC4 generates Java 17 compatible connectors and maintains backward compatibility only with 0.8.0-RC3 connectors built without Java customizations. 0.8.0-RC4 is not backward compatible with REST SDK versions earlier than 0.8.0-RC3. 
+
+When upgrading REST SDK connectors to Java 17:
+
+* For connectors with versions earlier than 0.8.0-RC3, or connectors with version 0.8.0-RC3 with Java customizations:
++
+Regenerate the connector with 0.8.0-RC4 and port over any Java customizations. The resulting connector will likely break compatibility with the previous version of the connector.
```

Comment:
And...? So it requires a new major version of the connector? Is there something they need to do when backward compatibility is broken?

Reply:
I think it is up to partners on how they version their connectors